test_framework/scripts/longhorn-setup.sh

```diff
@@ -25,20 +25,45 @@ set_kubeconfig_envvar(){
   ARCH=${1}
   BASEDIR=${2}
 
-    if [[ ${ARCH} == "amd64" ]] ; then
+  if [[ ${ARCH} == "amd64" ]] ; then
     if [[ ${TF_VAR_k8s_distro_name} == [rR][kK][eE] ]]; then
       export KUBECONFIG="${BASEDIR}/kube_config_rke.yml"
     elif [[ ${TF_VAR_k8s_distro_name} == [rR][kK][eE]2 ]]; then
       export KUBECONFIG="${BASEDIR}/terraform/aws/${DISTRO}/rke2.yaml"
+    elif [[ ${TF_VAR_k8s_distro_name} == "eks" ]]; then
+      export KUBECONFIG="${BASEDIR}/eks.yml"
     else
       export KUBECONFIG="${BASEDIR}/terraform/aws/${DISTRO}/k3s.yaml"
     fi
   elif [[ ${ARCH} == "arm64"  ]]; then
-    export KUBECONFIG="${BASEDIR}/terraform/aws/${DISTRO}/k3s.yaml"
+    if [[ ${TF_VAR_k8s_distro_name} == "eks" ]]; then
+      export KUBECONFIG="${BASEDIR}/eks.yml"
+    else
+      export KUBECONFIG="${BASEDIR}/terraform/aws/${DISTRO}/k3s.yaml"
+    fi
   fi
 }
 
 
+create_admin_service_account(){
+  kubectl apply -f "${TF_VAR_tf_workspace}/templates/kubeconfig_service_account.yaml"
+  TOKEN=$(kubectl -n kube-system get secret/kubeconfig-cluster-admin-token -o=go-template='{{.data.token}}' | base64 -d)
+  yq -i ".users[0].user.token=\"${TOKEN}\""  "${TF_VAR_tf_workspace}/eks.yml"
+}
+
+
+install_iscsi(){
+  kubectl apply -f "https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/prerequisite/longhorn-iscsi-installation.yaml"
+}
+
+
+install_cluster_autoscaler(){
```

Comment:
Suggest just installing from the manifest curl -O https://raw.githubusercontent.com/kubernetes/autoscaler/master/cluster-autoscaler/cloudprovider/aws/examples/cluster-autoscaler-autodiscover.yaml instead of creating cluster_autoscaler.yaml maintained by our own. 

The same for other public cloud distros.

Reply:
refined.